Files/View Models/ItemViewModel.cs

```diff
@@ -803,34 +784,50 @@ public async Task EnumerateItemsFromSpecialFolderAsync(string path)
                 FileSizeBytes = 0
             };
 
-            if (Connection != null)
+            await Task.Run(async () =>
             {
-                var value = new ValueSet();
-                value.Add("Arguments", "RecycleBin");
-                value.Add("action", "Enumerate");
-                // Send request to fulltrust process to enumerate recyclebin items
-                var response = await Connection.SendMessageAsync(value);
-                // If the request was canceled return now
-                if (_addFilesCTS.IsCancellationRequested)
-                {
-                    IsLoadingItems = false;
-                    return;
-                }
-                if (response.Status == AppServiceResponseStatus.Success
-                    && response.Message.ContainsKey("Enumerate"))
+                if (Connection != null)
                 {
-                    var folderContentsList = JsonConvert.DeserializeObject<List<ShellFileItem>>((string)response.Message["Enumerate"]);
-                    for (int count = 0; count < folderContentsList.Count; count++)
+                    var value = new ValueSet();
+                    value.Add("Arguments", "RecycleBin");
+                    value.Add("action", "Enumerate");
+                    // Send request to fulltrust process to enumerate recyclebin items
+                    var response = await Connection.SendMessageAsync(value);
+                    // If the request was canceled return now
+                    if (_addFilesCTS.IsCancellationRequested)
+                    {
+                        return;
+                    }
+                    if (response.Status == AppServiceResponseStatus.Success
+                        && response.Message.ContainsKey("Enumerate"))
                     {
-                        var item = folderContentsList[count];
-                        AddFileOrFolderFromShellFile(item, returnformat);
-                        if (count % 300 == 0)
+                        var folderContentsList = JsonConvert.DeserializeObject<List<ShellFileItem>>((string)response.Message["Enumerate"]);
+                        var tempList = new List<ListedItem>();
+                        for (int count = 0; count < folderContentsList.Count; count++)
                         {
-                            OrderFiles();
+                            var item = folderContentsList[count];
+                            var listedItem = AddFileOrFolderFromShellFile(item, returnformat);
+                            if (listedItem != null)
+                            {
+                                tempList.Add(listedItem);
+                            }
+                            if (count % 64 == 0)
```

Comment:
Did you change the number to 64 on purpose?

Reply:
No could be "300" i haven't tested if there's an improvement

Reply:
No, it could be "300". But since I wait at least X elements to begin showing them on the UI that number shouldn't be too high.
I'll test if there's any improvement one way or the other.

Reply:
My comment deleted as no longer relevant